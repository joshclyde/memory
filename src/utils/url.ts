export const CREATE_TAG_URL = `/tag/create`;
export const editTagByIdUrl = (id: string) => `/tag/edit/${id}`;
export const CREATE_FLASHCARD_URL = `/flashcard/create`;
export const editFlashcardByIdUrl = (id: string) => `/flashcard/edit/${id}`;
export const EDIT_FLASHCARD_URL = `/flashcard/edit`;
export const reviewTagUrl = (id: string) => `/review/${id}`;
export const learnTagUrl = (id: string) => `/learn/${id}`;
export const HOME_URL = `/`;
