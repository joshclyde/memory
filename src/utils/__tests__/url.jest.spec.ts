import { describe, expect, it } from "@jest/globals";
import * as urlUtils from "../url";

describe("url utils", () => {
  it.each(Object.entries(urlUtils))("%s", (_, url) => {
    if (typeof url === "string") {
      expect(url).toMatchSnapshot();
    } else {
      expect(url("foo")).toMatchSnapshot();
    }
  });
});
