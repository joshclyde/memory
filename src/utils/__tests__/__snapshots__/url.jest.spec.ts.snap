// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`url utils CREATE_FLASHCARD_URL 1`] = `"/flashcard/create"`;

exports[`url utils CREATE_TAG_URL 1`] = `"/tag/create"`;

exports[`url utils EDIT_FLASHCARD_URL 1`] = `"/flashcard/edit"`;

exports[`url utils HOME_URL 1`] = `"/"`;

exports[`url utils editFlashcardByIdUrl 1`] = `"/flashcard/edit/foo"`;

exports[`url utils editTagByIdUrl 1`] = `"/tag/edit/foo"`;

exports[`url utils learnTagUrl 1`] = `"/learn/foo"`;

exports[`url utils reviewTagUrl 1`] = `"/review/foo"`;
